nuxeo-retention-core/src/test/resources/retention-rules-contrib-test.xml

```diff
@@ -81,6 +81,19 @@
       <end-condition expression="">
       </end-condition>
     </rule>
+    
+    <rule>
+      <id>retentionOnCreationWithEndDate</id>
+      <begin-condition event="documentCreated"
+        expression="document.getType()=='File'">
+      </begin-condition>
+      <begin-delay />
+      <retention-end-date>dc:expired</retention-end-date> 
+      <begin-action>Document.Lock</begin-action>
+      <end-action>Document.Unlock</end-action>
+      <end-condition expression="">
```

Comment:
I actually remembered that the end-condition was supposed to be used the way you use 'retention-end-date'

Reply:
I understand and I tried to implement it but finally I stay on my implementation (end remove the end condition) because with a end-condition we are not able to fill the disposal date on the record object and it seemed to me more clear.